### ISSUE TITLE
Platform channel messages receive no reply in `testWidgets`, causing hard-to-debug test hangs

### ISSUE THREAD

**Issue by bramp** [flutter/flutter#134502]
### Is there an existing issue for this?

- [X] I have searched the [existing package issues](https://github.com/flutter/flutter/issues?q=is%3Aopen+is%3Aissue+label%3Apackage)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### What package does this bug report belong to?

path_provider

### What target platforms are you seeing this bug on?

macOS

### Have you already upgraded your packages?

Yes

### Dependency versions

<details><summary>pubspec.lock</summary>

```lock
# Generated by pub
# See https://dart.dev/tools/pub/glossary#lockfile
packages:
  async:
    dependency: transitive
    description:
      name: async
      sha256: "947bfcf187f74dbc5e146c9eb9c0f10c9f8b30743e341481c1e2ed3ecc18c20c"
      url: "https://pub.dev"
    source: hosted
    version: "2.11.0"
  boolean_selector:
    dependency: transitive
    description:
      name: boolean_selector
      sha256: "6cfb5af12253eaf2b368f07bacc5a80d1301a071c73360d746b7f2e32d762c66"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  characters:
    dependency: transitive
    description:
      name: characters
      sha256: "04a925763edad70e8443c99234dc3328f442e811f1d8fd1a72f1c8ad0f69a605"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.0"
  clock:
    dependency: transitive
    description:
      name: clock
      sha256: cb6d7f03e1de671e34607e909a7213e31d7752be4fb66a86d29fe1eb14bfb5cf
      url: "https://pub.dev"
    source: hosted
    version: "1.1.1"
  collection:
    dependency: transitive
    description:
      name: collection
      sha256: f092b211a4319e98e5ff58223576de6c2803db36221657b46c82574721240687
      url: "https://pub.dev"
    source: hosted
    version: "1.17.2"
  cupertino_icons:
    dependency: "direct main"
    description:
      name: cupertino_icons
      sha256: d57953e10f9f8327ce64a508a355f0b1ec902193f66288e8cb5070e7c47eeb2d
      url: "https://pub.dev"
    source: hosted
    version: "1.0.6"
  fake_async:
    dependency: transitive
    description:
      name: fake_async
      sha256: "511392330127add0b769b75a987850d136345d9227c6b94c96a04cf4a391bf78"
      url: "https://pub.dev"
    source: hosted
    version: "1.3.1"
  ffi:
    dependency: transitive
    description:
      name: ffi
      sha256: "7bf0adc28a23d395f19f3f1eb21dd7cfd1dd9f8e1c50051c069122e6853bc878"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.0"
  flutter:
    dependency: "direct main"
    description: flutter
    source: sdk
    version: "0.0.0"
  flutter_lints:
    dependency: "direct dev"
    description:
      name: flutter_lints
      sha256: a25a15ebbdfc33ab1cd26c63a6ee519df92338a9c10f122adda92938253bef04
      url: "https://pub.dev"
    source: hosted
    version: "2.0.3"
  flutter_test:
    dependency: "direct dev"
    description: flutter
    source: sdk
    version: "0.0.0"
  lints:
    dependency: transitive
    description:
      name: lints
      sha256: "0a217c6c989d21039f1498c3ed9f3ed71b354e69873f13a8dfc3c9fe76f1b452"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  matcher:
    dependency: transitive
    description:
      name: matcher
      sha256: "1803e76e6653768d64ed8ff2e1e67bea3ad4b923eb5c56a295c3e634bad5960e"
      url: "https://pub.dev"
    source: hosted
    version: "0.12.16"
  material_color_utilities:
    dependency: transitive
    description:
      name: material_color_utilities
      sha256: "9528f2f296073ff54cb9fee677df673ace1218163c3bc7628093e7eed5203d41"
      url: "https://pub.dev"
    source: hosted
    version: "0.5.0"
  meta:
    dependency: transitive
    description:
      name: meta
      sha256: "3c74dbf8763d36539f114c799d8a2d87343b5067e9d796ca22b5eb8437090ee3"
      url: "https://pub.dev"
    source: hosted
    version: "1.9.1"
  path:
    dependency: transitive
    description:
      name: path
      sha256: "8829d8a55c13fc0e37127c29fedf290c102f4e40ae94ada574091fe0ff96c917"
      url: "https://pub.dev"
    source: hosted
    version: "1.8.3"
  path_provider:
    dependency: "direct main"
    description:
      name: path_provider
      sha256: a1aa8aaa2542a6bc57e381f132af822420216c80d4781f7aa085ca3229208aaa
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  path_provider_android:
    dependency: transitive
    description:
      name: path_provider_android
      sha256: "6b8b19bd80da4f11ce91b2d1fb931f3006911477cec227cce23d3253d80df3f1"
      url: "https://pub.dev"
    source: hosted
    version: "2.2.0"
  path_provider_foundation:
    dependency: transitive
    description:
      name: path_provider_foundation
      sha256: "19314d595120f82aca0ba62787d58dde2cc6b5df7d2f0daf72489e38d1b57f2d"
      url: "https://pub.dev"
    source: hosted
    version: "2.3.1"
  path_provider_linux:
    dependency: transitive
    description:
      name: path_provider_linux
      sha256: f7a1fe3a634fe7734c8d3f2766ad746ae2a2884abe22e241a8b301bf5cac3279
      url: "https://pub.dev"
    source: hosted
    version: "2.2.1"
  path_provider_platform_interface:
    dependency: transitive
    description:
      name: path_provider_platform_interface
      sha256: "94b1e0dd80970c1ce43d5d4e050a9918fce4f4a775e6142424c30a29a363265c"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  path_provider_windows:
    dependency: transitive
    description:
      name: path_provider_windows
      sha256: "8bc9f22eee8690981c22aa7fc602f5c85b497a6fb2ceb35ee5a5e5ed85ad8170"
      url: "https://pub.dev"
    source: hosted
    version: "2.2.1"
  platform:
    dependency: transitive
    description:
      name: platform
      sha256: ae68c7bfcd7383af3629daafb32fb4e8681c7154428da4febcff06200585f102
      url: "https://pub.dev"
    source: hosted
    version: "3.1.2"
  plugin_platform_interface:
    dependency: transitive
    description:
      name: plugin_platform_interface
      sha256: da3fdfeccc4d4ff2da8f8c556704c08f912542c5fb3cf2233ed75372384a034d
      url: "https://pub.dev"
    source: hosted
    version: "2.1.6"
  sky_engine:
    dependency: transitive
    description: flutter
    source: sdk
    version: "0.0.99"
  source_span:
    dependency: transitive
    description:
      name: source_span
      sha256: "53e943d4206a5e30df338fd4c6e7a077e02254531b138a15aec3bd143c1a8b3c"
      url: "https://pub.dev"
    source: hosted
    version: "1.10.0"
  stack_trace:
    dependency: transitive
    description:
      name: stack_trace
      sha256: c3c7d8edb15bee7f0f74debd4b9c5f3c2ea86766fe4178eb2a18eb30a0bdaed5
      url: "https://pub.dev"
    source: hosted
    version: "1.11.0"
  stream_channel:
    dependency: transitive
    description:
      name: stream_channel
      sha256: "83615bee9045c1d322bbbd1ba209b7a749c2cbcdcb3fdd1df8eb488b3279c1c8"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.1"
  string_scanner:
    dependency: transitive
    description:
      name: string_scanner
      sha256: "556692adab6cfa87322a115640c11f13cb77b3f076ddcc5d6ae3c20242bedcde"
      url: "https://pub.dev"
    source: hosted
    version: "1.2.0"
  term_glyph:
    dependency: transitive
    description:
      name: term_glyph
      sha256: a29248a84fbb7c79282b40b8c72a1209db169a2e0542bce341da992fe1bc7e84
      url: "https://pub.dev"
    source: hosted
    version: "1.2.1"
  test_api:
    dependency: transitive
    description:
      name: test_api
      sha256: "75760ffd7786fffdfb9597c35c5b27eaeec82be8edfb6d71d32651128ed7aab8"
      url: "https://pub.dev"
    source: hosted
    version: "0.6.0"
  vector_math:
    dependency: transitive
    description:
      name: vector_math
      sha256: "80b3257d1492ce4d091729e3a67a60407d227c27241d6927be0130c98e741803"
      url: "https://pub.dev"
    source: hosted
    version: "2.1.4"
  web:
    dependency: transitive
    description:
      name: web
      sha256: dc8ccd225a2005c1be616fe02951e2e342092edf968cf0844220383757ef8f10
      url: "https://pub.dev"
    source: hosted
    version: "0.1.4-beta"
  win32:
    dependency: transitive
    description:
      name: win32
      sha256: "9e82a402b7f3d518fb9c02d0e9ae45952df31b9bf34d77baf19da2de03fc2aaa"
      url: "https://pub.dev"
    source: hosted
    version: "5.0.7"
  xdg_directories:
    dependency: transitive
    description:
      name: xdg_directories
      sha256: "589ada45ba9e39405c198fe34eb0f607cddb2108527e658136120892beac46d2"
      url: "https://pub.dev"
    source: hosted
    version: "1.0.3"
sdks:
  dart: ">=3.1.1 <4.0.0"
  flutter: ">=3.7.0"

```

</details>

### Steps to reproduce

1) Create this test:

```dart
  testWidgets('getApplicationSupportDirectory()', (WidgetTester tester) async {
    await getApplicationSupportDirectory();
  });
  ```

2) `flutter test`

3) Wait forever, no exception, no timeout, just hangs.

### Expected results

I get some kind of error, most likely `MissingPlatformDirectoryException(..)`

### Actual results

The test hangs forever.

### Code sample

<details><summary>Code sample</summary>

```shell
flutter create bug
cd bug
$EDITOR test/widget_test.dart
```

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:path_provider/path_provider.dart';

import 'package:bug/main.dart';

void main() {
  testWidgets('getApplicationSupportDirectory()', (WidgetTester tester) async {
    await getApplicationSupportDirectory();
  });
}

```

</details>

### Screenshots or Videos

N/A

### Logs

<details><summary>Logs</summary>

```console
❯ flutter test --verbose
[   +2 ms] executing: sw_vers -productName
[  +11 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[   +5 ms] Exit code 0 from: sw_vers -productVersion
[        ] 13.5.2
[        ] executing: sw_vers -buildVersion
[   +5 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 22G91
[        ] executing: uname -m
[   +2 ms] Exit code 0 from: uname -m
[        ] arm64
[   +2 ms] executing: sysctl hw.optional.arm64
[   +1 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[  +20 ms] Found 1 files which will be executed as Widget Tests.
[   +2 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +14 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'LegacyCanvasKitRemover' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +25 ms] Skipping pub get: version match.
[  +16 ms] Found plugin path_provider at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider-2.1.1/
[   +2 ms] Found plugin path_provider_android at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_android-2.2.0/
[   +1 ms] Found plugin path_provider_foundation at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_foundation-2.3.1/
[   +1 ms] Found plugin path_provider_linux at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_linux-2.2.1/
[   +1 ms] Found plugin path_provider_windows at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_windows-2.2.1/
[  +29 ms] Found plugin path_provider at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider-2.1.1/
[        ] Found plugin path_provider_android at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_android-2.2.0/
[        ] Found plugin path_provider_foundation at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_foundation-2.3.1/
[        ] Found plugin path_provider_linux at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_linux-2.2.1/
[        ] Found plugin path_provider_windows at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_windows-2.2.1/
[  +17 ms] Found plugin path_provider at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider-2.1.1/
[        ] Found plugin path_provider_android at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_android-2.2.0/
[        ] Found plugin path_provider_foundation at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_foundation-2.3.1/
[        ] Found plugin path_provider_linux at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_linux-2.2.1/
[        ] Found plugin path_provider_windows at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_windows-2.2.1/
[  +35 ms] Generating /Users/bramp/geocrew/flutter/bug/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +10 ms] executing: sysctl hw.optional.arm64
[   +2 ms] Exit code 0 from: sysctl hw.optional.arm64
[        ] hw.optional.arm64: 1
[        ] executing: /usr/bin/arch -arm64e xcrun xcodebuild -version
[ +124 ms] Exit code 0 from: /usr/bin/arch -arm64e xcrun xcodebuild -version
[        ] Xcode 14.3.1
           Build version 14E300c
[  +67 ms] shaderc command: [/opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/artifacts/engine/darwin-x64/impellerc, --sksl, --iplr,
--sl=build/unit_test_assets/shaders/ink_sparkle.frag, --spirv=build/unit_test_assets/shaders/ink_sparkle.frag.spirv,
--input=/opt/homebrew/Caskroom/flutter/3.13.2/flutter/packages/flutter/lib/src/material/shaders/ink_sparkle.frag, --input-type=frag,
--include=/opt/homebrew/Caskroom/flutter/3.13.2/flutter/packages/flutter/lib/src/material/shaders,
--include=/opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/artifacts/engine/darwin-x64/shader_lib]
[ +119 ms] running test package with arguments: [--chain-stack-traces, --, file:///Users/bramp/geocrew/flutter/bug/test/widget_test.dart]
00:00 +0: loading /Users/bramp/geocrew/flutter/bug/test/widget_test.dart                                                                                            [  +67 ms] test 0: starting test /Users/bramp/geocrew/flutter/bug/test/widget_test.dart
[   +2 ms] Stopping scan for flutter_test_config.dart; found project root at /Users/bramp/geocrew/flutter/bug
[        ] Compiler will use the following file as its incremental dill file:
/var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_compiler.iYkLpR/output.dill
[        ] Listening to compiler controller...
[   +3 ms] Compiling file:///var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_listener.SXlf5V/listener.dart
[   +4 ms] Found plugin path_provider at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider-2.1.1/
[        ] Found plugin path_provider_android at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_android-2.2.0/
[        ] Found plugin path_provider_foundation at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_foundation-2.3.1/
[        ] Found plugin path_provider_linux at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_linux-2.2.1/
[ +663 ms] Found plugin path_provider_windows at /Users/bramp/.pub-cache/hosted/pub.dev/path_provider_windows-2.2.1/
[  +12 ms] /opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/dart-sdk/bin/snapshots/frontend_server.dart.snapshot --sdk-root
/opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --no-print-incremental-dependencies
--target=flutter --experimental-emit-debug-metadata -DFLUTTER_WEB_AUTO_DETECT=true --output-dill
/var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_compiler.iYkLpR/output.dill --packages
/Users/bramp/geocrew/flutter/bug/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation
--initialize-from-dill /Users/bramp/geocrew/flutter/bug/build/test_cache/build/c075001b96339384a97db4862b8ab8db.cache.dill.track.dill --verbosity=error
[   +9 ms] <- compile file:///var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_listener.SXlf5V/listener.dart
00:01 +0: loading /Users/bramp/geocrew/flutter/bug/test/widget_test.dart                                                                                            [ +543 ms] <- accept
[        ] <- reset
[        ] Compiling file:///var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_listener.SXlf5V/listener.dart took 1233ms
[        ] test 0: starting test device
[   +3 ms] test 0: awaiting connection to test device
[        ] test 0: VM Service uri is not available
[   +1 ms] test 0: test harness socket server is running at port:53352
[        ] Using this directory for fonts configuration: /var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_fonts.bMjwB5
[        ] test 0: Starting flutter_tester process with command=[/opt/homebrew/Caskroom/flutter/3.13.2/flutter/bin/cache/artifacts/engine/darwin-x64/flutter_tester,
--disable-vm-service, --enable-checked-mode, --verify-entry-points, --enable-software-rendering, --skia-deterministic-rendering, --enable-dart-profiling,
--non-interactive, --use-test-fonts, --disable-asset-fonts, --packages=/Users/bramp/geocrew/flutter/bug/.dart_tool/package_config.json,
--flutter-assets-dir=/Users/bramp/geocrew/flutter/bug/build/unit_test_assets,
/var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_listener.SXlf5V/listener.dart.dill], environment={FLUTTER_TEST: true,
FONTCONFIG_FILE: /var/folders/hm/0wzzr0bx1s38dbcmzjgjrc1m0000gn/T/flutter_tools.nAvLnB/flutter_test_fonts.bMjwB5/fonts.conf, SERVER_PORT: 53352, APP_NAME: bug,
UNIT_TEST_ASSETS: /Users/bramp/geocrew/flutter/bug/build/unit_test_assets}
[   +3 ms] test 0: Started flutter_tester process at pid 25748
[ +237 ms] test 0: connected to test device, now awaiting test result
[        ] test 0: Waiting for test harness or tests to finish
00:49 +0: getApplicationSupportDirectory()
```

</details>

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
❯ flutter doctor -v
[✓] Flutter (Channel stable, 3.13.3, on macOS 13.5.2 22G91 darwin-arm64, locale en-GB)
    • Flutter version 3.13.3 on channel stable at /opt/homebrew/Caskroom/flutter/3.13.2/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 2524052335 (5 days ago), 2023-09-06 14:32:31 -0700
    • Engine revision b8d35810e9
    • Dart version 3.1.1
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
    • Android SDK at /Users/bramp/Library/Android/sdk
    • Platform android-34, build-tools 34.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b829.9-10027231)

[✓] VS Code (version 1.82.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.73.20230904

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 14 (API 34) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.5.2 22G91 darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 116.0.5845.179

[✓] Network resources
    • All expected network resources are available.

• No issues found!

```

</details>

**Comment by bramp** [comment]
Some extra notes. I discovered then, when using the google_fonts package in a test, and finding it would hang (because it was effectively waiting on getApplicationSupportDirectory).

Stackoverflow has also told me to write:

```dart
TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger
        .setMockMethodCallHandler(
            const MethodChannel('plugins.flutter.io/path_provider'),
            (MethodCall methodCall) async {
      return '.';
    });
```

and that does indeed fix the problem. However, I would have expected getApplicationSupportDirectory to throw an exception, not hang forever.

**Comment by huycozy** [comment]
Looks like you put the test file into wrong directory. There are already some written tests in the package here: [path_provider_test.dart](https://github.com/flutter/packages/blob/main/packages/path_provider/path_provider/test/path_provider_test.dart) that you may want to take a look. I just ran those tests (including `getApplicationSupportDirectory` test) and saw `All tests passed` as expected. 

For `testWidgets`, you can try putting the test file into `integration_test` and see if it also works (it worked from my end).

**Comment by stuartmorgan** [comment]
Please see https://docs.flutter.dev/testing/plugins-in-tests

Closing as expected behavior; plugins can't work in tests other than integration tests.

**Comment by bramp** [comment]
Thanks for the quick feedback.  Reviewing https://docs.flutter.dev/testing/plugins-in-tests it says

> plugins only work when your code is running in your application, such as with flutter run or when running [integration tests](https://docs.flutter.dev/cookbook/testing/integration/introduction). When running [Dart unit tests](https://docs.flutter.dev/cookbook/testing/unit/introduction) or [widget tests](https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html), the host code isn’t available. If the code you are testing calls any plugins, this often results in errors like the following: [..] MissingPluginException

What I'm observing is it just hanging forever. I would expect it to throw a MissingPluginException. 

Comparing my test to those in [path_provider_test.dart](https://github.com/flutter/packages/blob/main/packages/path_provider/path_provider/test/path_provider_test.dart), the difference is I'm using `testWidgets` vs `test`, when I switch to using `test` the issue goes away and I get the MissingPluginException.

So I still think there is a bug here, as testWidgets can be used in unit/widget tests, and hanging forever does not provide any information to the developer. (However I admit, this bug may be outside of the path_provider package).

**Comment by stuartmorgan** [comment]
`MissingPluginException` is just one example of a possible outcome of using plugins in non-integration tests, but it is odd that it's not giving at least a bindings error. @Hixie I know you changed some things about the test bindings; is silent no-reply the expected default outcome for a platform channel message in this context?

**Comment by bramp** [comment]
Aha, I figured it out. https://github.com/flutter/flutter/issues/76790. "general practice to use tester.runAsync() in your testWidget"

If I change my test to:

```
testWidgets('getApplicationSupportDirectory()', (WidgetTester tester) async {
    await tester.runAsync(() async {
      await getApplicationSupportDirectory();
    });
});
```

Now I get a MissingPluginException.  Something about testWidgets changes how async is handled in the tests, requiring this wrapper. This seems a little brittle and not well documented.

**Comment by bramp** [comment]
ah and https://github.com/flutter/flutter/issues/5728 seems to be about fixing this sharp edge.

thanks for investigating.

**Comment by goderbauer** [comment]
I am going to close this as a duplicate of https://github.com/flutter/flutter/issues/5728.
